app/connectors/MiddleConnector.scala

```diff
@@ -45,13 +48,21 @@ class MiddleConnector @Inject()(httpHandler: HttpHandler)(implicit appConfig: Ap
   def connectToAtsListOnBehalfOf(uar: Uar, requestedUTR: SaUtr)(implicit hc: HeaderCarrier): Future[AtsResponse] =
     connectToAtsList(requestedUTR)
 
-  def connectToPayeATS(nino: Nino, taxYear: Int)(implicit hc: HeaderCarrier): Future[HttpResponse] =
-    http.GET[HttpResponse](url("/taxs/" + nino + "/" + taxYear + "/paye-ats-data"))
+  def connectToPayeATS(nino: Nino, taxYear: Int)(
+    implicit hc: HeaderCarrier): Future[Either[UpstreamErrorResponse, HttpResponse]] =
+    http.GET[Either[UpstreamErrorResponse, HttpResponse]](url("/taxs/" + nino + "/" + taxYear + "/paye-ats-data")) recover handleHttpExceptions
 
   def connectToPayeATSMultipleYears(nino: Nino, yearFrom: Int, yearTo: Int)(
-    implicit hc: HeaderCarrier): Future[HttpResponse] =
-    http.GET[HttpResponse](url(s"/taxs/$nino/$yearFrom/$yearTo/paye-ats-data"))
+    implicit hc: HeaderCarrier): Future[Either[UpstreamErrorResponse, HttpResponse]] =
+    http.GET[Either[UpstreamErrorResponse, HttpResponse]](url(s"/taxs/$nino/$yearFrom/$yearTo/paye-ats-data")) recover handleHttpExceptions
+
+  def connectToGovernmentSpend(taxYear: Int)(
+    implicit hc: HeaderCarrier): Future[Either[UpstreamErrorResponse, HttpResponse]] =
+    http.GET[Either[UpstreamErrorResponse, HttpResponse]](url(s"/taxs/government-spend/$taxYear")) recover handleHttpExceptions
 
-  def connectToGovernmentSpend(taxYear: Int)(implicit hc: HeaderCarrier): Future[HttpResponse] =
-    http.GET[HttpResponse](url(s"/taxs/government-spend/$taxYear"))
+  val handleHttpExceptions: PartialFunction[Throwable, Either[UpstreamErrorResponse, HttpResponse]] = {
+    case e: HttpException =>
+      logger.error(e.message)
+      Left(UpstreamErrorResponse(e.message, e.responseCode))
```

Comment:
`e.responseCode` this is not the http status code, there is no status code for those error you have top make one up, 502 will do.